lib/types/json.js

```diff
@@ -109,9 +109,9 @@ function json(options) {
 
     // assert charset per RFC 7159 sec 8.1
     var charset = getCharset(req) || 'utf-8'
-    if (charset.substr(0, 4) !== 'utf-') {
+    if (['utf-8', 'utf-16', 'utf-32'].indexOf(charset) === -1) {
```

Comment:
Hi! Please undo this line, because we allow the charset names UTF-16LE and UTF-16BE, and this will now deny those two charsets :(